deploy/helm/sumologic/templates/instrumentation/traces-sampler/configmap.yaml

```diff
@@ -1,5 +1,19 @@
 {{ $tracesEnabled := .Values.sumologic.traces.enabled }}
 {{- if $tracesEnabled }}
+{{ $baseConfig := (tpl (.Files.Get "conf/instrumentation/traces.sampler.conf.yaml") .) | fromYaml }}
+{{ $mergeConfig := .Values.tracesSampler.config.merge }}
+{{ $overrideConfig := .Values.tracesSampler.config.override }}
+{{ $finalConfig := "" }}
+{{ if $overrideConfig }}
+{{ $finalConfig = $overrideConfig }}
+{{ else }}
+{{ $finalConfig = mergeOverwrite $baseConfig $mergeConfig }}
+{{ end }}
+{{ $valuesYamlConfig := .Values.tracesSampler.config }}
+{{ $useValuesYamlConfig := include "useDefaultConfig" $valuesYamlConfig | trim }}
+{{ if eq $useValuesYamlConfig "true" }}
+{{ $finalConfig = include "removeMergeAndOverrideKeys" .Values.tracesSampler.config }}
+{{ end }}
```

Comment:
Why is it needed? Can't we just treat all configs as overrided? Is it to ensure backward compatibility?